OpenFAST build errors on macOS missing MPI symbols when building as shared library

I'm seeing these errors during the 99% linking phase when building on macOS with OpenMPI 3.1.3:
```
Undefined symbols for architecture x86_64:
  "_MPI_Allreduce", referenced from:
      fast::OpenFAST::fillScInputsGlob()     in OpenFAST.cpp.o
  "_MPI_Comm_create", referenced from:
      fast::OpenFAST::allocateMemory()     in OpenFAST.cpp.o
  "_MPI_Comm_free", referenced from:
      fast::OpenFAST::end()    in OpenFAST.cpp.o
  "_MPI_Comm_group", referenced from:
      fast::OpenFAST::setInputs(fast::fastInputs const&) in OpenFAST.cpp.o
  "_MPI_Comm_rank", referenced from:
      fast::OpenFAST::allocateMemory()     in OpenFAST.cpp.o
      fast::OpenFAST::setInputs(fast::fastInputs const&) in OpenFAST.cpp.o
  "_MPI_Comm_size", referenced from:
      fast::OpenFAST::allocateTurbinesToProcsSimple()     in OpenFAST.cpp.o
  "_MPI_Group_free", referenced from:
      fast::OpenFAST::end()    in OpenFAST.cpp.o
  "_MPI_Group_incl", referenced from:
      fast::OpenFAST::allocateMemory()     in OpenFAST.cpp.o
  "_ompi_mpi_comm_null", referenced from:
      fast::OpenFAST::end()    in OpenFAST.cpp.o
      fast::OpenFAST::fillScInputsGlob()     in OpenFAST.cpp.o
      fast::OpenFAST::allocateMemory()     in OpenFAST.cpp.o
  "_ompi_mpi_double", referenced from:
      fast::OpenFAST::fillScInputsGlob()     in OpenFAST.cpp.o
  "_ompi_mpi_op_sum", referenced from:
      fast::OpenFAST::fillScInputsGlob()     in OpenFAST.cpp.o
ld: symbol(s) not found for architecture x86_64
```
I will update this when I test whether or not OpenMPI 1.10.4 still works, which was previously known to work.

--- a/glue-codes/openfast-cpp/CMakeLists.txt
+++ b/glue-codes/openfast-cpp/CMakeLists.txt
@@ -42,6 +42,7 @@ target_link_libraries(openfastcpplib
   ${HDF5_HL_LIBRARIES}
   ${ZLIB_LIBRARIES}
   ${LIBXML2_LIBRARIES}
+  ${MPI_LIBRARIES}
   ${CMAKE_DL_LIBS})
 
 add_executable(openfastcpp

